### ISSUE TITLE
Write CI tests for Julia

### ISSUE THREAD

**Issue by jabranham** [emacs-ess/ESS#902]
Would be nice so we don't break julia. It might be possible to run these on travis too, see #901. When we do this, we should skip R and julia tests if the program isn't installed.

**Comment by solna86** [comment]
Yes, in fact ess seems broken now in melpa-stable due to julia dependencies:

`Error (use-package): Failed to install ess: Package ‘julia-mode-0.3’ is unavailable
Error (use-package): Cannot load ess`

**Comment by jabranham** [comment]
That's because the julia-mode folks removed their release tags, so it doesn't show up in MELPA-stable. You'll have to get julia-mode from MELPA.

Related issue: https://github.com/JuliaEditorSupport/julia-emacs/issues/46

**Comment by solna86** [comment]
Thanks. I was using both ESS and julia-mode from MELPA, just trying MELPA stable to see if I could catch a little annoying bug introduced a few days ago. Namely that `M-p` breaks ESS inferior-mode for Julia. That is, retrieving previous commands messes up the formatting.

I guess I should simply stick to MELPA unstable or mix and pin packages from different repositories.

**Comment by jabranham** [comment]
> Thanks. I was using both ESS and julia-mode from MELPA, just trying MELPA stable to see if I could catch a little annoying bug introduced a few days ago. Namely that `M-p` breaks ESS inferior-mode for Julia. That is, retrieving previous commands messes up the formatting.

Please report bugs you find. They don't get fixed if they don't get reported :-)

That said, I just pushed a fix for some of the history stuff, so perhaps your issue is solved now?

> I guess I should simply stick to MELPA unstable or mix and pin packages from different repositories.

Yes, this is possible by modifying `package-pinned-packages`.

**Comment by solna86** [comment]
> Please report bugs you find. They don't get fixed if they don't get reported :-) That said, I just pushed a fix for some of the history stuff, so perhaps your issue is solved now?

Thanks. I thought it might be an issue on my side, so I was trying to dissect it first. Seems like a real bug, so reported in #908.
